src/modelfree/training/victim_envs.py

```diff
@@ -94,8 +112,65 @@ def get_curry_obs(self, env_idx=None):
             return self._obs[env_idx]
 
 
+class MultiCurryVecEnv(CurryVecEnv):
+    def __init__(self, venv, policies, agent_idx=0, deterministic=False):
+        """Fixes one of the players in a VecMultiEnv, but alternates between policies
+        used to perform response action.
+        :param venv(VecMultiEnv): the environments.
+        :param policies(iterable of Policy): the policies to use for the agent at agent_idx.
+        :param agent_idx(int): the index of the agent that should be fixed.
+        :param policy_selector(func): A function that takes in
+        :return: a new VecMultiEnv with num_agents decremented. It behaves like env but
+                 with all actions at index agent_idx set to those sampled from one policy
+                 within policies"""
+        super().__init__(venv, policies, agent_idx, deterministic)
+
+        self.policies = policies
+        self.num_envs = venv.num_envs
+        self.state_array = [None]*self.num_envs
+        self.current_policy_idx = [None]*self.num_envs
+        self.current_policies = [None]*self.num_envs
+        self.switch_policy()
+
+    def step_async(self, actions):
+        policy_actions = []
+        for i in range(len(self._obs)):
+            policy = self.current_policies[i]
+            observations = np.array([self._obs[i]]*self.num_envs)
+            state = np.array([self.state_array[i]]*self.num_envs)
+            mask = np.array([self._dones[i]]*self.num_envs)
+            action, new_state = policy.predict(observations,
+                                               state=state,
+                                               mask=mask,
+                                               deterministic=self.deterministic)
+            policy_actions.append(action[0])
+            self.state_array.append(new_state)
+        policy_actions_array = np.array(policy_actions)
+        actions.insert(self._agent_to_fix, policy_actions_array)
+        self.venv.step_async(actions)
+
+    def get_policy(self):
```

Comment:
Ick. We'll need to just change the type for this and return the whole list.

Reply:
Do you think the best thing to do here (imagining a world, per your earlier comment, where we merge MultiCurryVecEnv and CurryVecEnv) would be to return a list of [policy] in the single-policy case, and <arbitrary order of policies> in the multi-policy case? Or should we return a num_envs list of policies in all cases, with the same policy just being present at all indices in the single-policy case?

Reply:
I can't see any uses of this method, so leaning towards just deleting it. I'm guessing we originally implemented it for transparency but it doesn't seem needed with the current approach.